x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -82,20 +84,56 @@ public FieldPermissions getFieldPermissions() {
          *         then this means that there are no document level restrictions
          */
         @Nullable
-        public Set<BytesReference> getQueries() {
-            return queries;
+        public DocumentPermissions getDocumentPermissions() {
+            return documentPermissions;
+        }
+
+        public static IndexAccessControl scopedIndexAccessControl(IndexAccessControl indexAccessControl,
```

Comment:
I don't think the name of this method, or the names of the parameters make it obvious what it does.
_scoped_ is not a bad name for what's being implemented, but it's also not a totally obvious name, and I'm not even 100% sure which IAC I should pass in which parameter.
I think, at the very least it needs java-doc, but I do wonder whether there's something more we could do with the naming. Perhaps if it were an instance method rather than being static, it might be easier to describe?

Reply:
I wonder if `filter` as a instance method name would be better?